hummingbot/connector/exchange/chainring/chainring_exchange.py

```
@@ -0,0 +1,509 @@
+
+        order_result = await self._api_post(
+            path_url=CONSTANTS.ORDER_PATH_URL,
+            data=api_params,
+            is_auth_required=True,
+            limit_id=CONSTANTS.ALL_HTTP
+        )
+
+        if order_result.get('requestStatus') in {"Rejected"}:
+            raise IOError({"label": "ORDER_REJECTED", "message": "Order rejected.", "error": order_result.get("error")})
+        exchange_order_id = str(order_result["orderId"])
+
+        return exchange_order_id, self._time_synchronizer.time()
+
+    async def _place_cancel(self, order_id: str, tracked_order: InFlightOrder):
+        precision = chainring_utils.symbol_precision(await self.exchange_info(), tracked_order.base_asset)
+        market_id = await self.exchange_symbol_associated_to_pair(trading_pair=tracked_order.trading_pair)
+
+        api_params = {
+            "marketId": market_id,
+            "orderId": tracked_order.exchange_order_id,
+            "side": CONSTANTS.SIDE_BUY if tracked_order.trade_type is TradeType.BUY else CONSTANTS.SIDE_SELL,
+            "amount": str(chainring_utils.move_point_right(tracked_order.amount, precision)),
+            "nonce": chainring_utils.generate_order_nonce(),
+            "verifyingChainId": CONSTANTS.DEFAULT_CHAIN_IDS.get(self._domain)
+        }
+        signature = self._auth.sign_place_cancel(api_params, await self.exchange_info())
+        api_params["signature"] = signature
+
+        try:
+            await self._api_delete(
+                path_url=f"{CONSTANTS.ORDER_PATH_URL}/{tracked_order.exchange_order_id}",
+                data=api_params,
+                is_auth_required=True,
+                limit_id=CONSTANTS.ALL_HTTP
+            )
+        except Exception as e:
+            if isinstance(e, ContentTypeError):
+                #  aiohttp seems not able to handle 204 NoContent response
+                return True
+            # if self._is_order_not_found_during_cancelation_error(e):
```

Does this not work?

---

thanks for pointing, forgot to cleanup.
`_is_order_not_found_during_cancelation_error` is an abstract method to help bot to understand reason of failed request. During adding tests I discovered that I should not hide this error, it is handled on exchange base level and there is corresponding base level test for it.